src/three-components/SmoothControls.ts

```diff
@@ -306,17 +332,41 @@ export class SmoothControls extends EventDispatcher {
     this.setOrbit();
     // Prevent interpolation in the case that any target spherical values
     // changed (preserving OrbitalControls behavior):
-    this[$spherical].copy(this[$targetSpherical]);
+    this[$spherical].copy(this[$destSpherical]);
   }
 
   /**
-   * Resets the zoom sensitivity based on model scale, here represented by
-   * framedHeight.
-   * @param framedHeight
+   *
    */
-  updateFramedHeight(framedHeight: number) {
+  updateFraming(framedHeight: number, modelDepth: number, aspect: number) {
```

Comment:
Framing seems pretty central to camera controls to me. What was strange was that both controls.ts and SmoothControls.ts were manipulating the camera, and when I was working on them I was running into a lot of bugs because it was unclear when one was stomping on the other. Also they were passing a lot of values back and forth, which is an indicator to me that the concepts are not well isolated. If you want to go down the bike shed route, I might rename SmoothControls to CameraControls and controls to CameraAttributes to make the current separation more clear.

Reply:
These are the questions that are running through my mind as I consider this:

 - If I used `SmoothControls` in a Three.js application that had nothing to do with `<model-viewer>`, would "framing" have meaning to me?
 - If I replaced Three.js with another renderer (thus requiring a new version of `SmoothControls` that worked in terms of that renderer's API), would I have to re-implement the notion of framing entirely?
 - Can the concept of framing be implemented without direct access to the private implementation details of `SmoothControls`?

Framing as a concept is currently part of the `ModelScene`. This makes sense after a fashion. After all, what purpose does the scene have if not to set the stage for the viewer?

OTOH it may make sense to extract it from `ModelScene` and implement it in a new construct that integrates a `ModelScene` and `SmoothControls`. But, adding it directly into `SmoothControls` is conflating two concepts without clear justification. Yes, it's obvious that there is overlap in terms of values that are used in both cases. But it's not obvious that there is conceptual overlap (or API even access requirements) to justify moving the implementation here.

Reply:
Put this back too, but keeping camera ownership in one place.

Reply:
Thanks!

We spoke f2f about this topic. `SmoothControls` is responsible for manipulating the position (and now FOV) of the camera, while another entity (currently `ModelScene`) is responsible for determining the "framing" of the model by adjusting the frustum and repositioning the camera as needed.

Position / FOV manipulation is the domain of `SmoothControls`, so `ModelScene` should never perform those changes directly on the camera instance.

FOV sits at an intersection between these responsibilities. It seems like there is a lot of meat left on the bone for improvement in how the camera reference is manipulated by these different constructs.